src/Chapter3.hs

```diff
@@ -480,6 +509,50 @@ After defining the city, implement the following functions:
    and at least 10 living __people__ inside in all houses of the city in total.
 -}
 
+data Castle
+  = None
+  | Castle {castleName :: String}
+  | CastleWithWalls {castleName :: String}
+  deriving (Show)
+
+-- Why can't we use 1 | 2 | 3 | 4 here? 
+data HousePeople
+  = One
+  | Two
+  | Three
+  | Four
+  deriving (Show)
+
+data House = House {housePeople :: HousePeople} deriving (Show)
```

Comment:
After you learned about newtypes, you can notice that `House` can be a `newtype` here 🙂